*** ip_nat.c.orig	Sat Jul 21 10:17:22 2001
--- ip_nat.c	Tue Oct 23 15:17:36 2001
***************
*** 2069,2075 ****
  	for (; nat; nat = nat->nat_hnext[0]) {
  		nflags = nat->nat_flags;
  
! 		if ((!ifp || ifp == nat->nat_ifp) &&
  		    nat->nat_inip.s_addr == srcip &&
  		    nat->nat_oip.s_addr == dst.s_addr &&
  		    (((p == 0) && (flags == (nflags & IPN_TCPUDP)))
--- 2069,2076 ----
  	for (; nat; nat = nat->nat_hnext[0]) {
  		nflags = nat->nat_flags;
  
! 		//if ((!ifp || ifp == nat->nat_ifp) &&
! 		if ( 
  		    nat->nat_inip.s_addr == srcip &&
  		    nat->nat_oip.s_addr == dst.s_addr &&
  		    (((p == 0) && (flags == (nflags & IPN_TCPUDP)))
***************
*** 2091,2097 ****
  	nat = nat_table[0][hv];
  	for (; nat; nat = nat->nat_hnext[0]) {
  		nflags = nat->nat_flags;
! 		if (ifp && ifp != nat->nat_ifp)
  			continue;
  		if (!(nflags & IPN_TCPUDP))
  			continue;
--- 2092,2099 ----
  	nat = nat_table[0][hv];
  	for (; nat; nat = nat->nat_hnext[0]) {
  		nflags = nat->nat_flags;
! 		//if (ifp && ifp != nat->nat_ifp)
! 		if (ifp)
  			continue;
  		if (!(nflags & IPN_TCPUDP))
  			continue;
*** ip_nat.h.orig	Sat Oct 27 15:39:09 2001
--- ip_nat.h	Sat Oct 27 15:43:39 2001
***************
*** 26,32 ****
  #define	SIOCGNATL	_IOWR(r, 63, struct natlookup *)
  #endif
  
! #undef	LARGE_NAT	/* define this if you're setting up a system to NAT
  			 * LARGE numbers of networks/hosts - i.e. in the
  			 * hundreds or thousands.  In such a case, you should
  			 * also change the RDR_SIZE and NAT_SIZE below to more
--- 26,32 ----
  #define	SIOCGNATL	_IOWR(r, 63, struct natlookup *)
  #endif
  
! #define	LARGE_NAT	/* define this if you're setting up a system to NAT
  			 * LARGE numbers of networks/hosts - i.e. in the
  			 * hundreds or thousands.  In such a case, you should
  			 * also change the RDR_SIZE and NAT_SIZE below to more
***************
*** 50,58 ****
  #undef	RDR_SIZE
  #undef	NAT_TABLE_SZ
  #undef	HOSTMAP_SIZE	127
! #define	NAT_SIZE	2047
! #define	RDR_SIZE	2047
! #define	NAT_TABLE_SZ	16383
  #define	HOSTMAP_SIZE	8191
  #endif
  #ifndef	APR_LABELLEN
--- 50,58 ----
  #undef	RDR_SIZE
  #undef	NAT_TABLE_SZ
  #undef	HOSTMAP_SIZE	127
! #define	NAT_SIZE	163839
! #define	RDR_SIZE	163839
! #define	NAT_TABLE_SZ	174079
  #define	HOSTMAP_SIZE	8191
  #endif
  #ifndef	APR_LABELLEN
